### PR TITLE
Added some functions.

### PR DESCRIPTION
**Pull request by sahithyandev**

### CODE REVIEW

File: hypemaths/models/matrix.py [hunk 1]
```diff
@@ -8,13 +8,16 @@ def __init__(
             self, matrix: t.Union[int, float, list] = None, dims: tuple = None, fill: t.Union[int, float] = None
     ) -> None:
         if not matrix:
-            if not dims or fill:
+            if not dims or fill is not None:
```
Comment (janaSunrise) [inline hypemaths/models/matrix.py:11]:
```suggestion
            if not dims or fill:
```
Actually `not dims` auto checks if its None or not :) So we wont need it

File: hypemaths/models/matrix.py [hunk 2]
```diff
@@ -30,6 +33,80 @@ def dims(self) -> tuple:
     def __repr__(self) -> str:
         return f"{self.__class__.__name__}({self.matrix})"
 
+    def multiply_by_number(self, number: t.Union[int, float]) -> 'Matrix':
+        '''
+        Multiplies the current matrix with a number.
+        '''
```
Comment (janaSunrise) [inline hypemaths/models/matrix.py:39]:
```suggestion
    def multiply_by_number(self, number: t.Union[int, float]) -> "Matrix":
        """Multiplies the current matrix with a number.""" 
```
Let's use `"` as the standard quotes as it's more universal and also `'` is used as secondary ones 😁

File: hypemaths/models/matrix.py [hunk 3]
```diff
@@ -30,6 +33,80 @@ def dims(self) -> tuple:
...
+        if type(number) is not int:
+            raise TypeError(f"Expected `int` but got {type(number).__name__} ({number})")
```
Comment (janaSunrise) [inline hypemaths/models/matrix.py:41]:
```suggestion
        if not isinstance(number, int):
            raise TypeError(f"Expected `int` but got {type(number)}")
```
This is an easier way to handle types :)

File: hypemaths/models/matrix.py [hunk 4]
```diff
@@ -30,6 +33,80 @@ def dims(self) -> tuple:
...
+            for column_i in range(original_matrix.cols):
+                new_matrix[row_i][column_i] = number * original_matrix[row_i][column_i]
+
+        return new_matrix
+
+    def transpose(self) -> 'Matrix':
+        '''
+        Transposes the matrix
+        '''
+        # [A.cols, A.rows] is the dimension of the new matrix
```
Comment (janaSunrise) [inline hypemaths/models/matrix.py:56]:
```suggestion
    def transpose(self) -> 'Matrix':
        """Transposes the matrix."""
        # [A.cols, A.rows] is the dimension of the new matrix
```

File: hypemaths/models/matrix.py [hunk 5]
```diff
@@ -30,6 +33,80 @@ def dims(self) -> tuple:
...
+        new_matrix = Matrix(dims=[self.cols, self.rows], fill=0)
+
+        for row_i in range(self.rows):
+            for column_i in range(self.cols):
+                new_matrix[column_i][row_i] = self[row_i][column_i]
```
Comment (janaSunrise) [inline hypemaths/models/matrix.py:61]:
```suggestion
        for row in range(self.rows):
            for column in range(self.cols):
                new_matrix[column][row] = self[row][column]
```
A cleaner approach 😉

File: hypemaths/models/matrix.py [hunk 6]
```diff
@@ -30,6 +33,80 @@ def dims(self) -> tuple:
...
+        for i in range(A.rows):
+            for j in range(A.cols):
+                added_matrix[i][j] = A[i][j] + B[i][j]
+
+        return added_matrix
+
+    @staticmethod
+    def is_matrix(A: 'Matrix') -> bool:
+        # TODO try not to use hard code "Matrix" here
+        return type(A).__name__ == 'Matrix'
```
Comment (janaSunrise) [inline hypemaths/models/matrix.py:108]:
```suggestion
    def is_matrix(matrix: "Matrix") -> bool:
        return isinstance(matrix, Matrix)
```

File: hypemaths/models/matrix.py [hunk 7]
```diff
@@ -67,7 +144,18 @@ def _create_filled_matrix(dims: tuple, fill: t.Union[int, float]) -> list:
                 f"The fill value must be integer or float, but the given fill value is {type(fill)}."
             )
 
-        return [[fill] * dims[1]] * dims[0]
+        # Reason:
+        #   n = Matrix(dims=[2, 3], fill=0)
+        #   n[1][1] = 45
+        #   print(n) # Matrix([[0, 45, 0], [0, 45, 0]])
+        # to stop this from happening
```
Comment (janaSunrise) [inline hypemaths/models/matrix.py:151]:
```suggestion
        """
        Reason:
          n = Matrix(dims=[2, 3], fill=0)
          n[1][1] = 45
          print(n) # Matrix([[0, 45, 0], [0, 45, 0]])
         to stop this from happening
         """
```
A better way for longer comments and explanations 👍🏼

**Review by janaSunrise — CHANGES_REQUESTED** [review]
I have requested some changes that just shorten and use substitute methods for them, and Out of curiosity, have you pulled the upstream branch for the latest changes?

**Comment by sahithyandev** [comment]
Yes. I did pull from the upstream.

I will make the requested changes and push soon.

File: hypemaths/models/matrix.py (hunk 1 above)
Comment (sahithyandev) [reply hypemaths/models/matrix.py:11]: But it won't work with `fill=0` case

**Comment by janaSunrise** [comment]
Great :+1: Looking forward to it, I just requested them, because they can be shorted and standard 😅

File: hypemaths/models/matrix.py (hunk 1 above)
Comment (janaSunrise) [reply hypemaths/models/matrix.py:11]: Ohh, Alright then :)

**Review by janaSunrise — APPROVED** [review]
Everything looks great! 👍🏼

**Pull request merged**